README.md

```diff
@@ -24,14 +21,77 @@ If you have downloaded the repo without using the `--recurse-submodules` argumen
 git submodule update --init --recursive
 ```
 
+## coreMQTT Agent Library Configurations
+
+The MQTT Agent library exposes build configuration macros that are required for building the library.
```

Comment:
I find this whole section confusing, please simplify.  For example "the agent library uses the same configuration file as coreMQTT, with the addition of the configuration constants listed at the top of [link to file where they are]." - or  "use of the coreMQTT agent requires the configuration parameters listed at the top of [link to file] to be added to the coreMQTT configuration file".  Or whatever.  Would also recommend linking to the API docs that describe the parameters.

Reply:
Reworded based on your suggestion and linked to the proper doc page